READMEファイルを作成

tomo1107 opened the pull request:
# What
- タスク：データベース設計
- ER図を作成し、READMEにデータベース設計を記述しました。
ER図のGyazoURL : https://gyazo.com/2ca759ecc34a92827ba26c3065af1c49
# Why
テーブル内容を整理する為。

tomo1107 committed:
READMEファイルを作成

READMEファイルとER図を作成しました

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| Column          | Type     | Options                        |
+| ------          | -------- | ------------------------------ |
+| nickname        | string   | null: false                    |
+| email           | string   | null: false                    |
```
```ruby
| email           | string   | null: false                    |
```
重複したemailが保存されないよう、emailカラムには一意性制約を付けましょう！

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| ------          | -------- | ------------------------------ |
+| nickname        | string   | null: false                    |
+| email           | string   | null: false                    |
+| password        | string   | null: false                    |
```
```ruby
| password        | string   | null: false                    |
```
こちらのカラムはdeviseのgemを使用しての実装になるため、deviseが予め用意している
encrypted_passwordというカラム名に修正しましょう！

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| last_name       | string   | null: false                    |
+| kana_first_name | string   | null: false                    |
+| kana_last_name  | string   | null: false                    |
+| birth_year      | int      | null: false                    |
```
```ruby
| birth_year      | int      | null: false                    |
| birth_month     | int      | null: false                    |
| birth_day       | int      | null: false                    |
```
誕生日情報のカラムは一つだけ用意しておき、データ型は「date型」に修正しましょう。
理由としましてはdate型は年、月、日を１つのカラムで取得することができるためです。

tomo1107 replied:
ご指摘の通り、修正しました。

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| item_name        | string     | null: false                    |
+| description      | text       | null: false                    |
+| price            | int        | null: false                    |
+| category         | string     | null: false                    |
```
```ruby
| category         | string     | null: false                    |
| item_status      | string     | null: false                    |
| delivery_fee     | string     | null: false                    |
| delivery_from    | string     | null: false                    |
| date_of_delivery | int        | null: false                    |
```
商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリーのカラムはactive_hashを用いて実装の予定のため、カラム名に「_id」を付けましょう！
またactive_hashを用いて実装した場合、情報を数字で保存しますのでTypeをinteger型に修正しましょう！

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| prefecture  | string     | null: false                    |
+| city        | string     | null: false                    |
+| address     | string     | null: false                    |
+| building    | string     | null: false                    |
```
```ruby
| building    | string     | null: false                    |
```
今回の実装ではビル名は入力必須ではありませんのでnull: false は削除しておきましょう！

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+### Association
+- belongs_to :user
+- belongs_to :item
 
```
購入履歴を管理するテーブルは１つの住所を持っているという設計にしたいので、住所情報を管理しているテーブルとhas_oneでアソシエーションと組みましょう。

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+## delivery_informationsテーブル
+| Column      | Type       | Options                        |
+| ----------- | ---------- | ------------------------------ |
+| postal_code | int        | null: false                    |
```
```ruby
| postal_code | int        | null: false                    |
```
郵便番号の情報が保存されるカラムは「-」ハイフンが入力されることも想定される為、string型に修正しましょう！

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,65 @@
+| Column      | Type       | Options                        |
+| ----------- | ---------- | ------------------------------ |
+| postal_code | int        | null: false                    |
+| prefecture  | string     | null: false                    |
```
```ruby
| prefecture  | string     | null: false                    |
```
配送先の都道府県情報を保存するカラムもアクティブハッシュを用いて実装しますので、カラム名に「_id」を付け、integer型に修正しましょう！

shuki-yamamoto reviewed (commented):
データベース設計のレビュー依頼ありがとうございます！
8点コメントいたしましたので、ご確認をお願いいたします。

tomo1107 committed:
是正対応1

以下のご指摘に対して、是正対応を行いました
https://github.com/tomo1107/furima-35073/pull/1#pullrequestreview-629656653

tomo1107 commented:
> データベース設計のレビュー依頼ありがとうございます！
> 8点コメントいたしましたので、ご確認をお願いいたします。

8件のご指摘を修正しましたので、再レビューの程、よろしくお願いします！
ER図のURL：https://gyazo.com/f09ce4df4a8180ed86bf5cb49689c6cc

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| Column              | Type     | Options                        |
+| ------              | -------- | ------------------------------ |
+| nickname            | string   | null: false                    |
+| email               | string   | null: false, uniqueness: true  |
```
こちらunique:trueとしましょう。
(uniqueness:trueはモデルにバリデーションを記述する際の書き方となります。)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| delivery_fee_id     | int        | null: false                    |
+| delivery_from_id    | int        | null: false                    |
+| date_of_delivery_id | int        | null: false                    |
+| is_purchased        | boolean    | null: false                    |
```
購入されたか否かはpurchaseテーブルで確認できるため、こちらのカラムは不要となります！
`| is_purchased        | boolean    | null: false                    |`

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| Column        | Type       | Options                        |
+| -----------   | ---------- | ------------------------------ |
+| postal_code   | string     | null: false                    |
+| prefecture_id | int        | null: false                    |
```
都道府県のモデルは商品テーブルと同じものを使い回すことができるため、こちらも「delivery_from_id」としてあげましょう！
(または、どちらも「prefecture_id」としてもよいですね！)

s-aka-moto commented on README.md:
```diff
@@ -1,24 +1,64 @@
+| city          | string     | null: false                    |
+| address       | string     | null: false                    |
+| building      | string     |                                |
+| phone         | int        | null: false                    |
```
電話番号は0から始まることを考慮し、数字の文字列として実装しておきたいためstring型で実装しましょう！

s-aka-moto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

tomo1107 commented:
> レビュー依頼ありがとうございます！
> 修正点確認させていただきました。
> 追加でコメントいたしましたので、ご確認お願いします！

ご指摘ありがとうございます。
全てのご指摘を修正しましたので、再レビューの程、よろしくお願いします！
ER図のURL：https://gyazo.com/f351a68a7f9dca8bdf8dc44f30465aa3

tomo1107 committed:
是正対応1

以下のご指摘に対して、是正対応を行いました。
https://github.com/tomo1107/furima-35073/pull/1#pullrequestreview-629713763

tomo1107 committed:
Revert "是正対応1"

This reverts commit e5dfa24fc600f1c341a80b67a73de49aa639f561.

tomo1107 committed:
是正対応2

以下のご指摘に対して、是正対応を行いました。
https://github.com/tomo1107/furima-35073/pull/1#pullrequestreview-629713763

shuki-yamamoto commented on README.md:
```diff
@@ -1,24 +1,63 @@
+
+### Association
+- has_many :items
+- has_one  :purchase
```
```ruby
has_one  :purchase
```
１人のユーザーは複数の購入履歴を保持することができるという設計にしたいのでpurchaseテーブルとhas_manyのアソシエーションを組みましょう！

shuki-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
1点だけコメントいたしましたので、ご確認お願いいたします。

tomo1107 committed:
是正対応3

以下の指摘に対して、是正対応を行いました。
https://github.com/tomo1107/furima-35073/pull/1#pullrequestreview-629757183

tomo1107 commented:
> レビュー依頼ありがとうございます！
> 修正点確認させていただきました。
> 1点だけコメントいたしましたので、ご確認お願いいたします。

ご指摘ありがとうございます。
全てのご指摘を是正しましたので、再レビューの程、よろしくお願いします。
ER図のGyazoURL：https://gyazo.com/9db576a00feb11c07faea7f3ee173042

shuki-yamamoto reviewed (commented):
レビュー依頼ありがとうございます。
修正点確認させていただきました。
LGTMとさせていただきます。
実装お疲れさまでした！

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き頑張っていきましょう！
